python/data_structures/linked_list.py

```diff
@@ -38,6 +38,73 @@ def includes(self, value):
             current = current.next
         return False
 
+    def append(self, new_value):
+        """
+        Append stores passed in element value as new_value, and calls new_node method with new_value as arg
+        If self.head == None we are at the end of the linked list, and can save our new node here.
+        Else we save the value of self.head as held_value, so we dont lose the element stored there.
+        while we have values in the nodes, or .next does not = None, reassign held_value to held_value.next
+        and traverse to the next node, until we have self.node == None.
+        """
+        new_node = Node(new_value)
+        if self.head is None:
+            self.head = new_node
+            return
+        else:
+            held_value = self.head
+            while held_value.next is not None:
+                held_value = held_value.next
+            held_value.next = new_node
+
+    def insert_before(self, value, new_value):
+        """
+        make and store our new node
+        iterate through our list, looking for our before. Storing the element previous to it each time.
+        once we found the elem which equals our before, we can iterate through the list until we find its
+        previous elem, from there we can reassign next to our new node, and insert our new node, with a .next
+        equal to the before elem.
+
+        """
```

Comment:
Docstrings should be used to describe _what_ your method/function accomplishes, not _how_ it is accomplished.

Your docstring on 91, for `insert_after`, is actually a great example of a concise docstring.

 That said, it can be a good idea to also include a short blurb about efficiency, such as "Executes in O(n) time, scaling with the linked list node count."